src/test_l1_dsHdmiIn.c

```diff
@@ -453,29 +453,48 @@ void test_l1_dsHdmiIn_negative_dsHdmiInGetStatus(void) {
- * |04|Call dsHdmiInSelectPort() |dsHDMI_IN_PORT_2, bool: false, dsVideoPlane_PRIMARY, bool false| dsERR_NONE | Port 2 should be selected |
- * |05|Call dsHdmiInTerm() to ensure deinitialization | | dsERR_NONE | Clean up after test |
+ * |02|Get the number of input ports from the profile file
+ * |03|Call dsHdmiInSelectPort() |dsHDMI_IN_PORT_0, bool: false, dsVideoPlane_PRIMARY, bool false| dsERR_NONE | Port 0 should be selected |
+ * |04|Call dsHdmiInSelectPort() |dsHDMI_IN_PORT_1, bool: true, dsVideoPlane_SECONDARY, bool true| dsERR_NONE | Port 1 should be selected |
+ * |05|Call dsHdmiInSelectPort() |dsHDMI_IN_PORT_2, bool: false, dsVideoPlane_PRIMARY, bool false| dsERR_NONE | Port 2 should be selected |
+ * |06||Call dsHdmiInSelectPort() | dsERR_OPERATION_NOT_SUPPORTED | Loop through the ports where HDMI input is not supported
+ * |07|Call dsHdmiInTerm() to ensure deinitialization | | dsERR_NONE | Clean up after test |
  * 
  */
 void test_l1_dsHdmiIn_positive_dsHdmiInSelectPort(void) {
     gTestID = 9;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID); 
-    // Step 1: Initialize the HDMI input sub-system using dsHdmiInInit()
-    UT_ASSERT_EQUAL(dsHdmiInInit(), dsERR_NONE);
-
-    // Step 2: Call dsHdmiInSelectPort() to select Port 0
-    UT_ASSERT_EQUAL(dsHdmiInSelectPort(dsHDMI_IN_PORT_0, false, dsVideoPlane_PRIMARY, false), dsERR_NONE);
-
-    // Step 3: Call dsHdmiInSelectPort() to select Port 1
-    UT_ASSERT_EQUAL(dsHdmiInSelectPort(dsHDMI_IN_PORT_1, true, dsVideoPlane_SECONDARY, true), dsERR_NONE);
+    uint8_t numInputPorts;
 
-    // Step 4: Call dsHdmiInSelectPort() to select Port 2
-    UT_ASSERT_EQUAL(dsHdmiInSelectPort(dsHDMI_IN_PORT_2, false, dsVideoPlane_PRIMARY, false), dsERR_NONE);
-
-    // Step 5: Call dsHdmiInTerm() to ensure deinitialization
-    UT_ASSERT_EQUAL(dsHdmiInTerm(), dsERR_NONE);
+    // Step 1: Initialize the HDMI input sub-system using dsHdmiInInit()
+    UT_ASSERT_EQUAL_FATAL(dsHdmiInInit(), dsERR_NONE);
+
+    // Step 2: Get the number of input ports from the profile file
+    numInputPorts = UT_KVP_PROFILE_GET_UINT8("dsHdmiIn/HdmiInputPort/numberOfPorts");
+    if (numInputPorts > 0) {
+       
+        for (uint8_t i = dsHDMI_IN_PORT_0; i < numInputPorts; i++)
+        {
+             // Step 3: Call dsHdmiInSelectPort() to select Port 0
+             UT_ASSERT_EQUAL(dsHdmiInSelectPort(i, false, dsVideoPlane_PRIMARY, false), dsERR_NONE);
+
+             // Step 4: Call dsHdmiInSelectPort() to select Port 1
+             UT_ASSERT_EQUAL(dsHdmiInSelectPort(i, true, dsVideoPlane_SECONDARY, true), dsERR_NONE);
+
+             // Step 5: Call dsHdmiInSelectPort() to select Port 2
+             UT_ASSERT_EQUAL(dsHdmiInSelectPort(i, false, dsVideoPlane_PRIMARY, false), dsERR_NONE);
+       }
+    } else {
+       
+	for (uint8_t i =  numInputPorts; i <= dsHDMI_IN_PORT_MAX; i++)
```

Comment:
for (uint8_t i =  numInputPorts; i <= dsHDMI_IN_PORT_MAX; i++) - No need to check for max value.

you can add like below 
for (uint8_t i =  numInputPorts; i < dsHDMI_IN_PORT_MAX; i++)

Reply:
Done